graphql_compiler/schema_transformation/merge_schemas.py

```diff
@@ -0,0 +1,173 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+from collections import namedtuple
+from copy import deepcopy
+
+from graphql import build_ast_schema
+from graphql.language import ast as ast_types
+from graphql.language.printer import print_ast
+import six
+
+from .utils import SchemaNameConflictError, check_ast_schema_is_valid, get_query_type_name
+
+
+MergedSchemaDescriptor = namedtuple(
+    'MergedSchemaDescriptor', (
+        'schema_ast',  # Document, AST representing the merged schema
+        'name_to_schema_id',  # Dict[str, str], type name to id of the schema the type is from
+    )
+)
+
+
+def _basic_schema_ast(query_type):
+    """Create a basic AST Document representing a nearly blank schema.
+
+    The output AST contains a single query type, whose name is the input string. The query type
+    is guaranteed to be the second entry of Document definitions, after the schema definition.
+    The query type has no fields.
+
+    Args:
+        query_type: str, name of the query type for the schema
+
+    Returns:
+        Document, representing a nearly blank schema
+    """
+    blank_ast = ast_types.Document(
+        definitions=[
+            ast_types.SchemaDefinition(
+                operation_types=[
+                    ast_types.OperationTypeDefinition(
+                        operation='query',
+                        type=ast_types.NamedType(
+                            name=ast_types.Name(value=query_type)
+                        ),
+                    )
+                ],
+                directives=[],
+            ),
+            ast_types.ObjectTypeDefinition(
+                name=ast_types.Name(value=query_type),
+                fields=[],
+                interfaces=[],
+                directives=[],
+            ),
+        ]
+    )
+    return blank_ast
+
+
+def merge_schemas(schemas_dict):
+    """Check that input schemas do not contain conflicting definitions, then merge.
+
+    Merged schema will contain all type, interface, enum, scalar, and directive definitions
```

Comment:
Assuming the merged schema will also contain all unions defined in the input schemas as well? Also a grammar nit -- maybe add `The` at the beginning of the sentence?